crates/polars-plan/src/logical_plan/builder.rs

```diff
@@ -199,7 +210,10 @@ impl LogicalPlanBuilder {
         // We set the hive partitions of the first path to determine the schema.
         // On iteration the partition values will be re-set per file.
         if hive_partitioning {
-            file_info.init_hive_partitions(path.as_path())?;
+            if let Some(uri) = uri {
```

Comment:
I _think_ this is correct and hive only applies to cloud URLs? This is why it's better not to have "a string" as a datastructure, and explicitly separate local paths from cloud URLs in the type system.